webapp/src/main/resources/public/js/sdk/entity/TmTextUnit.js

```diff
@@ -28,9 +28,11 @@ export default class TmTextUnit {
     static toTmTextUnit(json) {
         let result = new TmTextUnit();
 
-        result.id = json.id;
-        result.name = json.name;
-        result.content = json.content;
+        if (json) {
```

Comment:
Adding as we don't want the frontend to fail if the json object is null or somehow serialized incorrectly.